### PR TITLE
Improve assets search & mappings endpoints

### PR DESCRIPTION
**Pull request by prettyirrelevant**
## Checklist
- [x] Add `is_custom_asset` flag to `/assets/mappings` response.
- [x] Remove threshold from Levenshtein assets search.

Case Study:
A custom asset's name is Very Old Building Located at XYZ

and I'm trying to search for this asset... I'd have to type Very Old Building Located at for it to return that particular asset which defeats the point of searching.

Whereas searching for Very Old or Very should include that specific asset.

### CODE REVIEW

File: rotkehlchen/globaldb/handler.py [hunk 1]
```diff
@@ -299,24 +299,31 @@ def get_assets_mappings(identifiers: List[str]) -> Dict[str, dict]:
         Given a list of asset identifiers, return a list of asset information(id, name, symbol)
         for those identifiers.
         May raise:
-        - InputError if one of the identifiers do not exist
+        - InputError if one of the identifiers do not exist or invalid asset type found.
```
Comment (nebolax) [inline rotkehlchen/globaldb/handler.py:301]:
```suggestion
        - InputError if one of the identifiers does not exist or invalid asset type found.
```

File: rotkehlchen/globaldb/handler.py [hunk 2]
```diff
@@ -299,24 +299,31 @@ def get_assets_mappings(identifiers: List[str]) -> Dict[str, dict]:
...
-                'WHERE ' + identifiers_query,
-                tuple(identifiers),
-            )
-            for entry in cursor:
-                result[entry[0]] = {'name': entry[1], 'symbol': entry[2]}
-                if entry[3] is not None:
-                    result[entry[0]].update({'evm_chain': ChainID.deserialize_from_db(entry[3]).serialize()})  # noqa: E501
-            if len(result) != len(identifiers):
-                raise InputError('One or more of the given identifiers could not be found in the database')  # noqa: E501
+        try:
```
Comment (nebolax) [inline rotkehlchen/globaldb/handler.py:305]: Why is this try except needed?

Comment (prettyirrelevant) [reply rotkehlchen/globaldb/handler.py:305]: `AssetType.deserialize_from_db()` raises DeserializationError

Comment (nebolax) [reply rotkehlchen/globaldb/handler.py:305]: But isn't it guaranteed to always exist due to a foreign key? https://github.com/rotki/rotki/blob/develop/rotkehlchen/globaldb/schema.py#L115

Comment (prettyirrelevant) [reply rotkehlchen/globaldb/handler.py]: Yeah yeah. Updated.

**Review by nebolax — APPROVED** [review]
LooksGoodToMe

**Review by LefterisJP — COMMENTED** [review]
What are the changes you made in the search for @prettyirrelevant? They don't seem to be related to addiging the `is_custom_asset` flag

**Comment by prettyirrelevant** [comment]
> What are the changes you made in the search for @prettyirrelevant? They don't seem to be related to adding the `is_custom_asset` flag

Yeah, didn't want to create a new pr for it. I could do that though if this isn't the right way

**Comment by LefterisJP** [comment]
> > What are the changes you made in the search for @prettyirrelevant? They don't seem to be related to adding the `is_custom_asset` flag
> 
> Yeah, didn't want to create a new pr for it. I could do that though if this isn't the right way

no don't move them in another PR, but then describe what you are doing that for, and write it in the PR description. I review a PR whose only description is `Add is_custom_asset flag to assets mapping response` and see lots of unrelated changes, I have no idea what to think or how to review them

**Comment by prettyirrelevant** [comment]
> > > What are the changes you made in the search for @prettyirrelevant? They don't seem to be related to adding the `is_custom_asset` flag
> > 
> > 
> > Yeah, didn't want to create a new pr for it. I could do that though if this isn't the right way
> 
> no don't move them in another PR, but then describe what you are doing that for, and write it in the PR description. I review a PR whose only description is `Add is_custom_asset flag to assets mapping response` and see lots of unrelated changes, I have no idea what to think or how to review them

Done.

**Review by LefterisJP — APPROVED** [review]
LGTM. Only test_xratescom.py::test_get_historical_xratescom_exchange_rates failed. I think this started faling yesterday. We need to look it up and fix

**Pull request merged**